### PR TITLE
21979 - On Test, The EFT linkage Modal Take a long time to load

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:*
https://github.com/bcgov/entity/issues/21979

*Description of changes:*
Move multiple outer joins to a subquery join, also change some of the outer joins into inner joins

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

**Review by Jxio — APPROVED** [review]

File: pay-api/src/pay_api/services/eft_short_names.py [hunk 1]
```diff
@@ -602,36 +603,55 @@ def get_search_query(cls, search_criteria: EFTShortnamesSearch, is_count: bool =
...
             else:
-                query = query.filter_conditionally(search_criteria.amount_owing, statement_summary_query.c.total_owing)
+                subquery = subquery.filter_conditionally(
+                    search_criteria.amount_owing, statement_summary_query.c.total_owing)
 
-        query = cls.get_link_state_filters(search_criteria, query)
-        query = query.filter(or_(CfsAccountModel.payment_method == PaymentMethod.EFT.value,
-                                 CfsAccountModel.id.is_(None)))
-        query = query.filter(
+        subquery = cls.get_link_state_filters(search_criteria, subquery)
```
Comment (seeker25) [inline pay-api/src/pay_api/services/eft_short_names.py:626]: these aren't quite right, they should be main queries that filter out the data not just the subquery

**Comment by seeker25** [comment]
Closed and reopened because I wasn't sure if this code works on a search basis, but it still took 8s to look up a single account id

File: pay-api/src/pay_api/resources/v1/eft_short_names.py [hunk 2]
```diff
@@ -50,12 +50,14 @@ def get_eft_shortnames():
     short_name_id = request.args.get('shortNameId', None)
     statement_id = request.args.get('statementId', None)
     account_id = request.args.get('accountId', None)
+    account_id_list = request.args.get('accountIdList', None)
```
Comment (seeker25) [inline pay-api/src/pay_api/resources/v1/eft_short_names.py:53]: removing

File: pay-api/src/pay_api/services/eft_short_names.py [hunk 3]
```diff
@@ -572,66 +571,103 @@ def get_search_query(cls, search_criteria: EFTShortnamesSearch, is_count: bool =
...
+            )
 
-        query = cls.get_link_state_filters(search_criteria, query)
-        query = query.filter(or_(CfsAccountModel.payment_method == PaymentMethod.EFT.value,
-                                 CfsAccountModel.id.is_(None)))
-        query = query.filter(
-            or_(PaymentAccountModel.payment_method == PaymentMethod.EFT.value, PaymentAccountModel.id.is_(None))
-        )
+        if search_criteria.state:
+            if EFTShortnameStatus.UNLINKED.value in search_criteria.state:
```
Comment (seeker25) [inline pay-api/src/pay_api/services/eft_short_names.py:651]: refactor

File: pay-api/src/pay_api/services/eft_short_names.py [hunk 4]
```diff
@@ -572,66 +572,103 @@ def get_search_query(cls, search_criteria: EFTShortnamesSearch, is_count: bool =
...
-        query = cls.get_link_state_filters(search_criteria, query)
-        query = query.filter(or_(CfsAccountModel.payment_method == PaymentMethod.EFT.value,
-                                 CfsAccountModel.id.is_(None)))
-        query = query.filter(
-            or_(PaymentAccountModel.payment_method == PaymentMethod.EFT.value, PaymentAccountModel.id.is_(None))
-        )
+            # Short name link filters
+            query = query.filter_conditionally(search_criteria.account_id,
+                                               subquery.c.auth_account_id,
+                                               is_like=search_criteria.allow_partial_account_id)
```
Comment (seeker25) [inline pay-api/src/pay_api/services/eft_short_names.py:632]: as requested

**Pull request merged**